src/main/java/edu/cornell/kfs/coa/document/validation/impl/SubAccountGlobalRule.java

```diff
@@ -93,4 +148,308 @@ public boolean checkSubAccountDetail(SubAccountGlobalDetail dtl) {
+        }
+
+        // set a flag if any of the three reporting fields are filled (this is separated just for readability)
+        if (StringUtils.isNotEmpty(newSubAccountGlobal.getFinancialReportChartCode()) || StringUtils.isNotEmpty(newSubAccountGlobal.getFinReportOrganizationCode()) || StringUtils.isNotEmpty(newSubAccountGlobal.getFinancialReportingCode())) {
+            anyReportingFieldsEntered = true;
+        }
+
+        // if any of the three reporting code fields are filled out, all three must be, or none
+        if (anyReportingFieldsEntered && !allReportingFieldsEntered) {
+            putGlobalError(KFSKeyConstants.ERROR_DOCUMENT_SUBACCTMAINT_RPTCODE_ALL_FIELDS_IF_ANY_FIELDS);
+            success &= false;
+        }
+
+        return success;
+    }
+    
+    /**
+     * Checks CG rules.
+     * 
+     * @param document
+     * @return true if valid, false otherwise
+     */
+    protected boolean checkCgRules(MaintenanceDocument document) {
+    	boolean success = true;
+    	SubAccountGlobal newSubAccountGlobal = (SubAccountGlobal) super.getNewBo();
+    	List<SubAccountGlobalDetail> subAccountGlobalDetails = newSubAccountGlobal.getSubAccountGlobalDetails();
+    	
+    	for(SubAccountGlobalDetail subAccountGlobalDetail : subAccountGlobalDetails){
+    		success &= checkCgRules(newSubAccountGlobal, subAccountGlobalDetail );
+    	}
+    	return success;
+    }
+    
+    /**
+     * Checks to make sure that if cgAuthorized is false it succeeds immediately, otherwise it checks that all the information
+     * for CG is correctly entered and identified including:
+     * <ul>
+     * <li>If the {@link SubFundGroup} isn't for Contracts and Grants then check to make sure that the cost share and ICR fields are
+     * not empty</li>
+     * <li>If it isn't a child of CG, then the SubAccount must be of type ICR</li>
+     * </ul>
+     * 
+     * @param document
+     * @return true if the user is not authorized to change CG fields, otherwise it checks the above conditions
+     */
+    protected boolean checkCgRules(SubAccountGlobal newSubAccountGlobal,  SubAccountGlobalDetail subAccountGlobalDetail) {
+
+        boolean success = true;
+
+        // short circuit if the parent account is NOT part of a CG fund group
+        boolean a21SubAccountRefreshed = false;
+        subAccountGlobalDetail.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+        if (ObjectUtils.isNotNull(subAccountGlobalDetail.getAccount())) {
+        	Account account = subAccountGlobalDetail.getAccount();
+        	account.refreshReferenceObject(KFSPropertyConstants.SUB_FUND_GROUP); 
+            if (ObjectUtils.isNotNull(account.getSubFundGroup())) {
+
+                // compare them, exit if the account isn't for contracts and grants
+                if (!getSubFundGroupService().isForContractsAndGrants(account.getSubFundGroup())) {
+
+                    if (checkCgCostSharingIsEmpty(subAccountGlobalDetail) == false) {
```

Comment:
It would be better to use "!" instead of "== false" here and in other places in this same file. Not sure if it's worth changing at this point unless other updates are needed.